pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -66,9 +82,21 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 	desiredMinor := getEffectiveMinor(releaseContext.DesiredVersion)
 
 	// if there is no difference in the minor version (4.y.z where 4.y is the same for current and desired), then we can still upgrade
+	// if no cluster overrides have been set
 	if currentMinor == desiredMinor {
 		klog.V(4).Infof("Precondition %q passed: minor from the current %s matches minor from the target %s (both %s).", pf.Name(), cv.Status.History[0].Version, releaseContext.DesiredVersion, currentMinor)
-		return nil
+		if condition := ClusterVersionOverridesCondition(clusterVersion); condition != nil {
+			klog.V(4).Infof("Update from %s to %s blocked by %s: %s", cv.Status.History[0].Version, releaseContext.DesiredVersion, condition.Reason, condition.Message)
```

Comment:
Can we test this with `--to-image`?  I'd expect `releaseContext.DesiredVersion` to be an empty string in that case until we fix [rhbz#1822513][1].  I'm not clear on what we'd do here before that bug gets fixed, but we should know what we're getting ourselves into in case we want to hold this on a fix for the `--to-image` bug.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1822513

Reply:
Yes, this is dependent on where we end up with [rhbz#1822513](https://bugzilla.redhat.com/show_bug.cgi?id=1822513), i.e. `releaseContext.DesiredVersion` will be an empty string.

Reply:
Further analysis of [rhbz#1822513](https://bugzilla.redhat.com/show_bug.cgi?id=1822513) shows it to be an issue with pulling incorrect history entry as current version (see https://bugzilla.redhat.com/show_bug.cgi?id=1822513#c14). That fix is independent of this fix although both of course are needed.

Reply:
Going to hold this until #394 fixes the `cv.Status.History[0].Version` approach to use completed updates instead.

/hold

Reply:
Fixed in [b72e843](https://github.com/openshift/cluster-version-operator/pull/364/commits/b72e843974ad165f33417da7addeaf4a76a2b690)

Reply:
/hold cancel